store/store.go

```diff
@@ -440,7 +472,7 @@ func (s *Store[H]) flushLoop() {
 		// don't flush and continue if pending batch is not grown enough,
 		// and Store is not stopping(headers == nil)
 		if s.pending.Len() < s.Params.WriteBatchSize && headers != nil {
-			continue
+			return
```

Comment:
Ah okay so you want to leave not large enough un-flushed batch in waiting and exit flush loop so that we aren't writing anything to disk for the `Sync` call, I see.